src/index.js

```diff
@@ -44,74 +55,72 @@ const create = (privateKey, value, seq, lifetime, callback) => {
  * @param {string} value value to be stored in the record.
  * @param {number} seq number representing the current version of the record.
  * @param {string} expiration expiration datetime for record in the [RFC3339]{@link https://www.ietf.org/rfc/rfc3339.txt} with nanoseconds precision.
- * @param {function(Error, entry)} [callback]
+ * @returns {Promise<IpnsEntry>} entry
  */
-const createWithExpiration = (privateKey, value, seq, expiration, callback) => {
+const createWithExpiration = (privateKey, value, seq, expiration) => {
   const validityType = ipnsEntryProto.ValidityType.EOL
-  _create(privateKey, value, seq, expiration, validityType, callback)
+  return _create(privateKey, value, seq, expiration, validityType)
 }
 
-const _create = (privateKey, value, seq, isoValidity, validityType, callback) => {
-  sign(privateKey, value, validityType, isoValidity, (error, signature) => {
-    if (error) {
-      log.error('record signature creation failed')
-      return callback(Object.assign(new Error('record signature verification failed'), { code: ERRORS.ERR_SIGNATURE_CREATION }))
-    }
+const _create = async (privateKey, value, seq, isoValidity, validityType) => {
+  const signature = await sign(privateKey, value, validityType, isoValidity)
 
-    const entry = {
-      value: value,
-      signature: signature,
-      validityType: validityType,
-      validity: isoValidity,
-      sequence: seq
-    }
+  const entry = {
+    value: value,
+    signature: signature,
+    validityType: validityType,
+    validity: isoValidity,
+    sequence: seq
+  }
 
-    log(`ipns entry for ${value} created`)
-    return callback(null, entry)
-  })
+  log(`ipns entry for ${value} created`)
+  return entry
 }
 
 /**
  * Validates the given ipns entry against the given public key.
  *
```

Comment:
```suggestion
 * @async
```

Reply:
There are other places missing `@async` js doc

Reply:
I think that jsdoc will automatically detect this. In their [docs](http://usejsdoc.org/tags-async.html) it says:
> In general, you do not need to use this tag, because JSDoc automatically detects asynchronous functions and identifies them in the generated documentation. However, if you are writing a virtual comment for an asynchronous function that does not appear in your code, you can use this tag to tell JSDoc that the function is asynchronous.

Reply:
so, it is strange that the tag exists, but fair enough, we can go with this!